### PR TITLE
Upgrade to make compatible with JupyterLab 3

### LINKED ISSUE #23: Update for compatibility with JupyterLab 3
### Problem

The current stable version of JupyterLab is 3.x, which is incompatible with the telemetry package. 

### Proposed Solution

Update dependencies and code to make it compatible with JupyterLab 3.

- [x] Update jupyterlab dependencies to 3.3.x
- [x] Replace phosphor dependencies with lumino
- [x] Update typescript to 4.8.1
- [x] Update dev dependencies
- [x] Update code for successful build

### PR DESCRIPTION
**Pull request by 3coins**
Fixes #23

- [x] Update jupyterlab dependencies to 3.3.x
- [x] Replace phosphor dependencies with lumino
- [x] Update typescript to 4.8.1
- [x] Update dev dependencies
- [x] Update code for successful build
- [x]  Add unit tests
- [x]  Fix linting

### CODE REVIEW

**Comment by fcollonval** [comment]
Thanks @3coins 

It will be good to add some testing for this one - at least the default linter/packaging/installation available from the cookiecutter. The best would of course be to have some tests that demonstrate its usage.

**Comment by 3coins** [comment]
> It will be good to add some testing for this one - at least the default linter/packaging/installation available from the cookiecutter. The best would of course be to have some tests that demonstrate its usage.

Will do. Thanks.

**Comment by blink1073** [comment]
Kicking now that actions are working again

**Comment by blink1073** [comment]
Ah, they'll only run in your fork until we merge.

**Comment by fcollonval** [comment]
@3coins it seems you have a syntax error in the workflow file: https://github.com/3coins/jupyterlab-telemetry/actions/runs/1994732900/workflow

**Comment by blink1073** [comment]
@3coins you accidentally checked in the `coverage` directory.

**Comment by 3coins** [comment]
@fcollonval @blink1073 
Thanks for catching the workflow error. I am still working on the tests, will correct this error shortly.

**Comment by 3coins** [comment]
@Zsailer
I am not able to get the python tests to pass, event though I can verify that the server extension is installed and tested that the endpoint works in jupyterlab, when I run pytest, it is returning a 404.
```
==================================================================== FAILURES =====================================================================
____________________________________________________________________ test_put _____________________________________________________________________

jp_fetch = <function jp_fetch.<locals>.client_fetch at 0x7faba2c279d0>

    async def test_put(jp_fetch):
>       r = await jp_fetch(
                    'telemetry',
                    method='PUT',
                    body=json.dumps({
                        'foo': 'bar'
                    })
                )
E       tornado.httpclient.HTTPClientError: HTTP 404: Not Found

jupyterlab_telemetry/tests/test_handlers.py:6: HTTPClientError
-------------------------------------------------------------- Captured stderr setup --------------------------------------------------------------
[D 2022-03-18 10:45:39.163 ServerApp] Config changed: {'ServerApp': {'jpserver_extensions': {'jupyter_telemetry': True}}, 'NotebookNotary': {'db_file': ':memory:'}}
The module 'jupyter_telemetry' could not be found. Are you sure the extension is installed?

WARNING  ServerApp:log.py:59 404 PUT /a%40b/telemetry (127.0.0.1) 25.35ms referer=None
============================================================= short test summary info =============================================================
FAILED jupyterlab_telemetry/tests/test_handlers.py::test_put - tornado.httpclient.HTTPClientError: HTTP 404: Not Found
```

Here is the server extension list showing that the extension is enabled.

```
± |jlab-3-upgrade U:1 ✗| → jupyter server extension list
Config dir: /Users/pijain/.jupyter

Config dir: /Users/pijain/opt/anaconda3/envs/jlab-3.x-telemetry/etc/jupyter
    jupyterlab_telemetry enabled
    - Validating jupyterlab_telemetry...
      jupyterlab_telemetry 1.0.0 OK
```
I have committed all the changes, but marked the test to skip.

**Comment by 3coins** [comment]
Fixed the python tests in the latest commit.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
+name: Build
+
+on:
+  push:
+    branches: main
```
Comment (fcollonval) [inline .github/workflows/build.yml]:
```suggestion
    branches: master
```

File: .github/workflows/check-release.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
+name: Check Release
+on:
+  push:
+    branches:
+      - main
```
Comment (fcollonval) [inline .github/workflows/check-release.yml]:
```suggestion
      - master
```

File: .github/workflows/check-release.yml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  pull_request:
+    branches:
+      - main
```
Comment (fcollonval) [inline .github/workflows/check-release.yml]:
```suggestion
      - master
```

**Review by fcollonval — CHANGES_REQUESTED** [review]
Thanks @3coins (especially for adding the tests)

I have a minor change as the main branch is called `master` on this repository.

Otherwise we should merge and iterate from there - in particular if the CI fails.

**Comment by blink1073** [comment]
Indeed both workflows failed

**Comment by 3coins** [comment]
> Indeed both workflows failed

Oh no, let me take a look.

**Pull request merged**